completions/fish/meson.build

```diff
@@ -0,0 +1,9 @@
+fish_completion_dir = get_option('fishcompletiondir')
+if fish_completion_dir = ''
+    fish_completion_dir = join_paths(get_option('datadir'), 'fish', 'completions')
```

Comment:
It appears fish doesn't look in `/usr/local/share/fish/completions`, which is what this evaluates to by default.

Do you think fish *should* look there (in which case would you mind opening a bug against them?), or should we pick a different directory by default?

Reply:
Fish sets an actual pkg config variable for where to put completions; so pulling it in as an optional dependency and setting that would the *correct* way to do it e.g.
```
fish_completion_dir = get_option('fishcompletiondir')
fish = dependency('fish', required : false)

if fish_completion_dir == '' and fish.found()
    fish_completion_dir = fish.get_pkgconfig_variable('completionsdir')
elif fish_completion_dir == ''
    fish_completion_dir = join_paths(get_option('datadir'), 'fish', 'completions')
endif
```

Reply:
Sounds good

Reply:
I read https://fishshell.com/docs/current/index.html#completion-path, and it recommends installing third-party completions to `/usr/share/fish/vendor_completions.d/`. So it sounds like if `fish.get_pkgconfig_variable('completionsdir')` fails, we should default to `datadir/fish/vendor_completions.d/` instead of `datadir/fish/completions/`. What do you think?

Reply:
Yeah that sounds good, but we should probably simply hard code `/usr/share/fish/vendor_completions.d/` since `datadir/fish/completions/` when meson's `--prefix` isn't set to `usr` it wont work.

Reply:
Alright, let's do that, with a comment that fish ignores the proper (`/usr/local/...`-based) location.